### DISCUSSION TITLE
FindMolChiralCenters can not detect s and r for a molecule while it can detect for substructure of this molecule

### DISCUSSION THREAD

**Discussion by gokhantahil** [rdkit/rdkit#6567] [Q&A]
Hello,

For the SMILES of `C1CC2(C1)CO[C@@H](CC2)CC(=O)N[C@H]3CC[C@@H](CC3)[NH2+]CCF`, the `FindMolChiralCenters(mol, force=True,includeUnassigned=True, useLegacyImplementation=False)` returns [(6, 'S')], which means it can only detect the first [C@@H]. However, the second and third carbon atoms with @ and @@ can not be detected. 

When I try to get the chiral centers of the `N[C@H]1CC[C@@H](CC1)[NH2+]CCF` it returns [(1, 'r'), (4, 'r')] with the same function. 

Why it happens, do I make a mistake or is it just a potential bug?

**Comment by gokhantahil** [comment]
After further inspections, it looks like the issue coming from `FindPotentialStereo` function which find only one candidate in the abowe SMILES:

```
from rdkit import Chem

mol = "C1CC2(C1)CO[C@@H](CC2)CC(=O)N[C@H]3CC[C@@H](CC3)[NH2+]CCF"
mol = Chem.MolFromSmiles(mol)
potentials = (Chem.FindPotentialStereo(mol))
for potential in potentials:
        print(potential.centeredOn)
```
        
returns 6. 

And the same code returns "1, 4" when changing the mol object as `mol = "N[C@H]1CC[C@@H](CC1)[NH2+]CCF"`

**Reply by bp-kelley** [reply]
This does seem odd to me, if I assign CIP labels I get reasonable results though:

```
>>> Chem.AssignCIPLabels(m)
>>> Chem.FindMolChiralCenters(m)
[(6, 'S'), (13, 'r'), (16, 'S')]
```

Going back to the original molecule, it looks like FindMolChiralCenters is not labeling ring stereo:

````
>>> for atom in m.GetAtoms():
...    d = atom.GetPropsAsDict()
...    del d['__computedProps']
...    print(d)
... 
{'_CIPRank': 0}
{'_CIPRank': 1}
{'_CIPRank': 7}
{'_CIPRank': 1}
{'_CIPRank': 11}
{'_CIPRank': 17}
{'_CIPRank': 12, '_ChiralityPossible': 1, '_CIPCode': 'S'}
{'_CIPRank': 5}
{'_CIPRank': 2}
{'_CIPRank': 6}
{'_CIPRank': 13}
{'_CIPRank': 18}
{'_CIPRank': 15}
{'_CIPRank': 9, '_ringStereochemCand': True, '_ringStereoAtoms': <rdkit.rdBase._vecti object at 0x11ca82d40>}
{'_CIPRank': 3}
{'_CIPRank': 4}
{'_CIPRank': 10, '_ringStereochemCand': True, '_ringStereoAtoms': <rdkit.rdBase._vecti object at 0x11ca82d40>}
{'_CIPRank': 4}
{'_CIPRank': 3}
{'_CIPRank': 16}
{'_CIPRank': 8}
{'_CIPRank': 14}
{'_CIPRank': 19}
```

So this might just be a bug.

**Reply by gokhantahil** [reply]
Correct me if i’m wrong but CIP Labels are not enough/correct as well since the expected result should be [(6, 'S'), (13, 'r'), (16, 'r')].

**Reply by bp-kelley** [reply]
I think the RDKit is correct here.
<img width="591" alt="image" src="https://github.com/rdkit/rdkit/assets/9888416/47347879-8f72-4b2b-a078-99109a07fa24">

**Reply by gokhantahil** [reply]
In this case, is this normal for the SMILES ```N[C@H]1CC[C@@H](CC1)[NH2+]CCF``` to get `[(1, 'r'), (4, 'r')]` with the same function?

```
mol = "N[C@H]1CC[C@@H](CC1)[NH2+]CCF"
m = Chem.MolFromSmiles(mol)
Chem.AssignCIPLabels(m)
print(Chem.FindMolChiralCenters(m))
```
returns 
`[(1, 'r'), (4, 'r')]`

**Comment by apahl** [comment]
My two cents: `FindMolChiralCenters` gives the correct answer of the compound having 1 chiral center, as the the two stereogenic centers at the cyclohexane are not chiral, they are "[pseudo-asymmetric](https://goldbook.iupac.org/terms/view/P04921)" ([question on chemistry stackexchange](https://chemistry.stackexchange.com/questions/153711/does-1-4-dimethylcyclohexane-have-a-chiral-center)).
